tests/syft/lib/python/complex/complex_test.py

```diff
@@ -0,0 +1,825 @@
+            Complex,
+            {1: 2},
+            1,
+        )
+        self.assertRaisesRegex(
+            TypeError,
+            "second argument must be a number, not 'dict'",
+            Complex,
+            1,
+            {1: 2},
+        )
+        # the following three are accepted by Python 2.6
+        self.assertRaises(ValueError, Complex, "1..1j")
+        self.assertRaises(ValueError, Complex, "1.11.1j")
+        self.assertRaises(ValueError, Complex, "1e1.1j")
+
+        # check that Complex accepts long unicode strings
+        self.assertEqual(type(Complex("1" * 500)), Complex)
+        # check whitespace processing
+        self.assertEqual(Complex("\N{EM SPACE}(\N{EN SPACE}1+1j ) "), 1 + 1j)
+        # Invalid unicode string
+        # See bpo-34087
+        self.assertRaises(ValueError, Complex, "\u3053\u3093\u306b\u3061\u306f")
+
+        class EvilExc(Exception):
+            pass
+
+        class evilcomplex:
+            def __complex__(self):
+                raise EvilExc
+
+        self.assertRaises(EvilExc, Complex, evilcomplex())
+
+        class float2:
+            def __init__(self, value):
+                self.value = value
+
+            def __float__(self):
+                return self.value
+
+        self.assertAlmostEqual(Complex(float2(42.0)), 42)
+        self.assertAlmostEqual(Complex(real=float2(17.0), imag=float2(23.0)), 17 + 23j)
+        self.assertRaises(TypeError, Complex, float2(None))
+
+        # FIXME doesn't work
+        # class MyIndex:
+        #     def __init__(self, value):
+        #         self.value = value
+
+        #     def __index__(self):
+        #         return self.value
+
+        # self.assertAlmostEqual(Complex(MyIndex(42)), 42.0 + 0.0j)
+        # self.assertAlmostEqual(Complex(123, MyIndex(42)), 123.0 + 42.0j)
+        # self.assertRaises(OverflowError, Complex, MyIndex(2 ** 2000))
+        # self.assertRaises(OverflowError, Complex, 123, MyIndex(2 ** 2000))
+
+        class MyInt:
+            def __int__(self):
+                return 42
+
+        self.assertRaises(TypeError, Complex, MyInt())
+        self.assertRaises(TypeError, Complex, 123, MyInt())
+
+        class complex0(Complex):
+            """Test usage of __complex__() when inheriting from 'Complex'"""
+
+            def __complex__(self):
+                return 42j
+
+        class complex1(Complex):
+            """Test usage of __complex__() with a __new__() method"""
+
+            def __new__(self, value=0j):
+                return Complex.__new__(self, 2 * value)
+
+            def __complex__(self):
+                return self
+
+        class complex2(Complex):
+            """Make sure that __complex__() calls fail if anything other than a
+            Complex is returned"""
+
+            def __complex__(self):
+                return None
+
+        self.assertEqual(Complex(complex0(1j)), 42j)
+        # Only deprecated from python 3.7
```

Comment:
We could add this:
```diff
+import sys
 import unittest

         self.assertEqual(Complex(complex0(1j)), 42j)
-        # Only deprecated from python 3.7
-        # with self.assertWarns(DeprecationWarning):
-        #     self.assertEqual(Complex(complex1(1j)), 2j)
+
+        if sys.version_info >= (3, 7):
+            # Only deprecated from python 3.7
+            with self.assertWarns(DeprecationWarning):
+                self.assertEqual(Complex(complex1(1j)), 2j)
         self.assertRaises(TypeError, Complex, complex2(1j))
```

Reply:
Indeed, I chose the too easy solution! I'll add that